webapp/channels/src/components/advanced_text_editor/advanced_text_editor.tsx

```diff
@@ -409,6 +410,14 @@ const AdvancedTextEditor = ({
         }
     }, [showPreview]);
 
+    // Focus textbox when selectedPostFocussedAt changes
+    useEffect(() => {
+        if (selectedPostFocussedAt) {
+            focusTextbox();
+        }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
This doesn't need to be changed now, but I am getting a bit worried about this rule normalizing either ignoring warnings or leading to normalizing disable comments. If anyone finds any good suggestions for replacement ESLint rules, let me know 😅

Reply:
React is adding "useEffectEvent", which... I don't like: https://react.dev/reference/react/useEffect#reading-the-latest-props-and-state-from-an-effect

It feels like solving a "problem you have created" with more boilerplate. But seems to be the direction React is moving.

Reply:
Yeah, I like that it makes it possible to fix this problem, but it continues making React less intuitive and less self-explanatory. It's the same problem I have with other reactive programming things like RxJS where they're so difficult to understand because all of the terminology is jargon.

I like some parts of `useEffect`, but I do wish they could go back in the direction of `componentDidUpdate` where it's easy to write "if X changed, do Y" without having to worry as much about where Y comes from

Reply:
That is how I read useEffect, and how I feel it should be read:

If any of the elements on the dependency list changes, do this.

That is why I also like to keep effects separated by purpose, and not trying to fit too much logic in a single effect (as it happens with component did update).

Reply:
> That is why I also like to keep effects separated by purpose, and not trying to fit too much logic in a single effect (as it happens with component did update).

Oh yeah, that's a massive advantage of `useEffect` since it really helps for encapsulating and letting you reuse logic which isn't just rendered. I wouldn't be surprised if that ability to break stuff down like that is a major reason for hooks in general since it feels like it brings back mixins from early React back before components were ES5 classes.

I just wish it was easier to grok the terminology and what's bound so that you could separate out what causes the `useEffect` to be triggered from what up to date values it needs